rust/src/io/local.rs

```diff
@@ -83,4 +88,15 @@ impl ObjectReader for LocalObjectReader {
         })
         .await?
     }
+
+    #[cfg(windows)]
+    async fn get_range(&self, range: Range<usize>) -> Result<Bytes> {
+        let file = self.file.clone();
+        tokio::task::spawn_blocking(move || {
+            let mut buf = BytesMut::zeroed(range.len());
+            file.seek_read(buf.as_mut(), range.start as u64)?;
```

Comment:
is this line the only difference? if so, how about to use the feature flag just over this line?

```rust

async fn get_range(...)  {
...
#[cfg(unix)]
file.read_at(buf.as_mut(), range.start as u64)
#[cfg(windows)
fie.seek_read(buf.as_mut(), range.start as u64)
...

}
```

It would make things easier to fix the rest of the function.

Reply:
Did not know you could do that, i thought the #[cfg()] macro was for methods only, theres a cfg! macro but I thought there was weirdness with conditional compilation, I'll give this approach a try though.

Reply:
you were correct, force pushed!